enum.rb

```diff
@@ -0,0 +1,43 @@
+# my_enumerable.rb
+module MyEnumerable
+  def all?
+    each do |element|
+      return false unless yield(element)
+    end
+    true
+  end
+
+  def any?
+    each do |element|
+      return true if yield(element)
+    end
+    false
+  end
+
+  def filter
+    result = []
+    each do |element|
+      result << element if yield(element)
+    end
+    result
+  end
+end
+
+# my_list.rb
+class MyList
+  include MyEnumerable
+  def initialize(*list)
+    @list = list
+  end
+
+  def each(&block)
+    @list.each(&block)
+  end
+end
```

Comment:
- [ ] According to the requirements Each class and module should have a separate .rb file.